graph/tarjan.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @function tarjan
+ * @description Given a graph, find the strongly connected components(SCC). A set of nodes form a SCC if there is a path between all pairs of points within that set.
+ * @Complexity_Analysis
+ * Time complexity: O(V + E). We perform a DFS of (V + E)
+ * Space Complexity: O(V). We hold numerous structures all of which at worst holds O(V) nodes.
+ * @param {[number, number][][]} graph - The graph in adjacency list form
+ * @return {number[][]} - An array of SCCs, where an SCC is an array with the indices of each node within that SCC.
```

Comment:
The SCCs are topologically sorted even, no?

Reply:
I think the opposite is true actually. Its reverse-topologically sorted (child SCCs are pushed to the array before its parent). I think Kosajaru returns them topologically sorted.

I'll mention this in the docs.